test/e2e/flask/solana-standard/connect.spec.ts

```diff
@@ -0,0 +1,297 @@
+          await connectSolanaTestDapp(driver, testDapp);
+
+          const header = await testDapp.getHeader();
+
+          const connectionStatus = await header.getConnectionStatus();
+          assert.strictEqual(
+            connectionStatus,
+            'Connected',
+            'Connection status should be "Connected"',
+          );
+
+          const account = await header.getAccount();
+          assert.strictEqual(
+            account,
+            '4tE7...Uxer',
+            'Connection status should be "4tE7...Uxer"',
+          );
+        },
+      );
+    });
+
+    it('Should be able to cancel connection and connect again', async function () {
+      await withSolanaAccountSnap(
+        {
+          ...DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+          title: this.test?.fullTitle(),
+        },
+        async (driver) => {
+          const testDapp = new TestDappSolana(driver);
+          await testDapp.openTestDappPage();
+
+          // 1. Start connection and cancel it
+          const header = await testDapp.getHeader();
+          await header.connect();
+
+          // wait to display wallet connect modal
+          await driver.delay(regularDelayMs);
+
+          const modal = await testDapp.getWalletModal();
+          await modal.connectToMetaMaskWallet();
+
+          await driver.delay(regularDelayMs);
+
+          // Cancel the connection
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+          await driver.clickElement({ text: 'Cancel', tag: 'button' });
+          await testDapp.switchTo();
+
+          // Verify we're not connected
+          const connectionStatus = await header.getConnectionStatus();
+          assert.strictEqual(
+            connectionStatus,
+            'Not connected',
+            'Connection status should be "Not connected" after cancellation',
+          );
+
+          // 2. Connect again
+          await connectSolanaTestDapp(driver, testDapp);
+
+          // Verify successful connection
+          const connectionStatusAfterConnect =
+            await header.getConnectionStatus();
+          assert.strictEqual(
+            connectionStatusAfterConnect,
+            'Connected',
+            'Connection status should be "Connected" after reconnecting',
+          );
+
+          const account = await header.getAccount();
+          assert.strictEqual(
+            account,
+            '4tE7...Uxer',
+            'Account should be "4tE7...Uxer"',
+          );
+        },
+      );
+    });
+  });
+
+  describe('Disconnect the dapp', function () {
+    it('Should disconnect', async function () {
+      await withSolanaAccountSnap(
+        {
+          ...DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+          title: this.test?.fullTitle(),
+        },
+        async (driver) => {
+          const testDapp = new TestDappSolana(driver);
+          await testDapp.openTestDappPage();
+          await connectSolanaTestDapp(driver, testDapp);
```

Comment:
rather than connecting via the dapp, can we setup a pre-existing connection using the fixtures? I'm not sure if that is possible for snaps.

Reply:
No reason this should be impossible for Solana. Agreed this would make sense in this test, but wouldn't block on this